### PR TITLE
Adding a replication controller for all monitoring pods.

### LINKED ISSUE #2408: monitoring example needs to run under a replication controller
examples/monitoring/influx-grafana-pod.json and friends NEED to be replaced by a replication controller ASAP.  As it stands they are bad examples.

Also, they should probably move to contrib/monitoring to match contrib/logging and contrib/dns

### PR DESCRIPTION
**Pull request by vishh**
Added a grafana service
Removed auth for Grafana to facilitate usage via service proxy on the api-server.
Removed elasticsearch dependency for monitoring - faster startup times.

Fixes #2408

### CODE REVIEW

File: cluster/addons/cluster-monitoring/heapster-controller.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  replicas: 1
+  replicaSelector:    
+    name: "heapster"
+  podTemplate:
+    desiredState:
+      manifest:
+        version: "v1beta1"
+        id: "monitoring-heapsterController"
+        containers:
+          -
```
Comment (lavalamp) [inline cluster/addons/cluster-monitoring/heapster-controller.yaml]: Can't this be combined with the following line?

File: cluster/gce/util.sh [hunk 2]
```diff
@@ -732,35 +732,33 @@ function setup-monitoring {
...
-      > "${KUBE_TEMP}/influx-grafana-pod.json"
     local kubectl="${KUBE_ROOT}/cluster/kubectl.sh"
-    if "${kubectl}" create -f "${KUBE_TEMP}/influx-grafana-pod.json" &> /dev/null \
-        && "${kubectl}" create -f "${KUBE_ROOT}/cluster/addons/cluster-monitoring/influx-grafana-service.json" &> /dev/null \
-        && "${kubectl}" create -f "${KUBE_ROOT}/cluster/addons/cluster-monitoring/heapster-pod.json" &> /dev/null; then
-      local dashboard_url="http://$(${kubectl} get -o json pod influx-grafana | grep hostIP | awk '{print $2}' | sed 's/[,|\"]//g')"
+    if "${kubectl}" create -f "${KUBE_ROOT}/cluster/addons/cluster-monitoring/influxdb-grafana-controller.yaml" &> /dev/null \
+      && "${kubectl}" create -f "${KUBE_ROOT}/cluster/addons/cluster-monitoring/influxdb-service.yaml" &> /dev/null \
+      && "${kubectl}" create -f "${KUBE_ROOT}/cluster/addons/cluster-monitoring/heapster-controller.yaml" &> /dev/null \
+      && "${kubectl}" create -f "${KUBE_ROOT}/cluster/addons/cluster-monitoring/grafana-service.yaml" &> /dev/null; then
```
Comment (lavalamp) [inline cluster/gce/util.sh]: I believe you can now pass the entire directory to kubectl and it will make all at the same time?

Comment (vishh) [inline cluster/gce/util.sh]: Done.

File: cluster/addons/cluster-monitoring/heapster-controller.yaml (hunk 1 above)
Comment (vishh) [inline cluster/addons/cluster-monitoring/heapster-controller.yaml]: Done.

**Comment by vishh** [comment]
@lavalamp addressed your comments and I improved the bash script also in the process. PTAL!

File: cluster/gce/util.sh [hunk 3]
```diff
@@ -732,35 +732,28 @@ function setup-monitoring {
...
-        "${KUBE_ROOT}/cluster/addons/cluster-monitoring/influx-grafana-pod.json" \
-      > "${KUBE_TEMP}/influx-grafana-pod.json"
     local kubectl="${KUBE_ROOT}/cluster/kubectl.sh"
-    if "${kubectl}" create -f "${KUBE_TEMP}/influx-grafana-pod.json" &> /dev/null \
-        && "${kubectl}" create -f "${KUBE_ROOT}/cluster/addons/cluster-monitoring/influx-grafana-service.json" &> /dev/null \
-        && "${kubectl}" create -f "${KUBE_ROOT}/cluster/addons/cluster-monitoring/heapster-pod.json" &> /dev/null; then
-      local dashboard_url="http://$(${kubectl} get -o json pod influx-grafana | grep hostIP | awk '{print $2}' | sed 's/[,|\"]//g')"
+    if "${kubectl}" create -f "${KUBE_ROOT}/cluster/addons/cluster-monitoring/" &> /dev/null; then
+      # wait for pods to be scheduled on a node.
+      while [[ $("${kubectl}" get pods -l name=influxGrafana -o template -t {{range.items}}{{.currentState.hostIP}}:{{end}}) == *"<"* ]]; do
```
Comment (satnam6502) [inline cluster/gce/util.sh]: Might this loop forever if something goes wrong? Timeout or a max number of attempts?

**Comment by satnam6502** [comment]
A great idea to use replication controllers here and I plan to do the same for the cluster logging setup although I might wait for `run` to be implemented.

File: cluster/gce/util.sh (hunk 3 above)
Comment (vishh) [inline cluster/gce/util.sh]: Done.

**Comment by vishh** [comment]
@satnam6502 PTAL

**Pull request merged**